### PR TITLE
Pah 223 offline

### LINKED ISSUE #223: check application offline working
Service worker is not installing correctly on production build. Investigation around it is required as when offline application won't open

### PR DESCRIPTION
**Pull request by jacol12345**
Closes #223

### CODE REVIEW

**Comment by jacol12345** [comment]
I will add description a bit later.

File: frontend/src/registerServiceWorker.js [hunk 1]
```diff
@@ -10,20 +10,17 @@ if (process.env.NODE_ENV === 'production') {
...
-    //   console.log('Content has been cached for offline use.');
-    // },
-    // offline() {
-    //   console.log('No internet connection found. App is running in offline mode.');
-    // },
-
-    // error(error) {
-    //   console.error('Error during service worker registration:', error);
-    // },
+    ready() {
```
Comment (adamgajzlerowicz) [inline frontend/src/registerServiceWorker.js]: probably just remove?

Comment (jacol12345) [reply frontend/src/registerServiceWorker.js]: I forced the SW to automatically refresh so we can get rid of the Refresh component.

File: frontend/public/service-worker.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  // Skip everything that is not a GET request.
+  if (event.request.method !== 'GET') {
+    event.respondWith(fetch(event.request));
+    return;
+  }
+
+  // For cached URLs, try to fetch first and fallback to cache
+  if (
+    cachedUrls.some(url => event.request.url.endsWith(url))
+    || cachedApiEndpoints.some(url => event.request.url.endsWith(url))
```
Comment (adamgajzlerowicz) [inline frontend/public/service-worker.js]: point free syntax please

**Comment by adamgajzlerowicz** [comment]
Please change target branch to develop

**Comment by jacol12345** [comment]
I added comments to `service-worker.js` to document how it works.

**Comment by adamgajzlerowicz** [comment]
How can I make a full production build locally to test this? Is make start enough or is there a different command to run this?
@jbj

**Comment by adamgajzlerowicz** [comment]
fixed with in different pr

**Pull request closed without merging**